app/scripts/controllers/transactions/tx-state-manager.test.js

```diff
@@ -253,85 +263,215 @@ describe('TransactionStateManager', function () {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x2',
-          },
-          status: TRANSACTION_STATUSES.APPROVED,
         },
         {
           id: 2,
+          status: TRANSACTION_STATUSES.UNAPPROVED,
+          txParams: { from: VALID_ADDRESS, to: VALID_ADDRESS_TWO },
           metamaskNetworkId: currentNetworkId,
-          chainId: currentChainId,
-          time: 2,
-          txParams: {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x2',
-          },
-          status: TRANSACTION_STATUSES.APPROVED,
         },
-      ];
-
-      const failedTx3s = [
         {
           id: 3,
+          status: TRANSACTION_STATUSES.UNAPPROVED,
+          txParams: { from: VALID_ADDRESS_TWO, to: VALID_ADDRESS },
           metamaskNetworkId: currentNetworkId,
-          time: 3,
-          txParams: {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x3',
-          },
-          status: TRANSACTION_STATUSES.FAILED,
         },
         {
-          id: 3,
+          id: 4,
+          status: TRANSACTION_STATUSES.UNAPPROVED,
+          txParams: { from: VALID_ADDRESS_TWO, to: VALID_ADDRESS },
           metamaskNetworkId: currentNetworkId,
-          chainId: currentChainId,
-          time: 3,
-          txParams: {
-            from: '0xAddress',
-            to: '0xRecipient',
-            nonce: '0x3',
-          },
-          status: TRANSACTION_STATUSES.FAILED,
         },
-      ];
-
-      const txm = new TxStateManager({
-        initState: {
-          transactions: [
-            ...submittedTx0s,
-            unapprovedTx1,
-            ...approvedTx2s,
-            ...failedTx3s,
-          ],
+        {
+          id: 5,
+          status: TRANSACTION_STATUSES.CONFIRMED,
+          txParams: { from: VALID_ADDRESS, to: VALID_ADDRESS_TWO },
+          metamaskNetworkId: currentNetworkId,
         },
-        getNetwork: () => currentNetworkId,
-        getCurrentChainId: () => currentChainId,
-      });
+        {
+          id: 6,
+          status: TRANSACTION_STATUSES.CONFIRMED,
+          txParams: { from: VALID_ADDRESS, to: VALID_ADDRESS_TWO },
+          metamaskNetworkId: currentNetworkId,
+        },
+        {
+          id: 7,
+          status: TRANSACTION_STATUSES.CONFIRMED,
+          txParams: { from: VALID_ADDRESS_TWO, to: VALID_ADDRESS },
+          metamaskNetworkId: currentNetworkId,
+        },
+        {
+          id: 8,
+          status: TRANSACTION_STATUSES.CONFIRMED,
+          txParams: { from: VALID_ADDRESS_TWO, to: VALID_ADDRESS },
+          metamaskNetworkId: currentNetworkId,
+        },
+        {
+          id: 9,
+          status: TRANSACTION_STATUSES.CONFIRMED,
+          txParams: { from: VALID_ADDRESS_TWO, to: VALID_ADDRESS },
+          metamaskNetworkId: currentNetworkId,
+        },
+      ];
+      txMetas.forEach((txMeta) => txStateManager.addTransaction(txMeta, noop));
```

Comment:
Nit: `addTransaction` doesn't take a second parameter

```suggestion
      txMetas.forEach((txMeta) => txStateManager.addTransaction(txMeta));
```

Reply:
There are a bunch more instances of this later as well

Reply:
Thanks! Resolved in latest!